File: tests/mocks/ui_mock.rb
# frozen_string_literal: true

# Mock version of UI class
class UIMock
  attr_accessor :echos, :error_msm

  def initialize
    clear
  end

  def echo(msg, color = nil)
    @echos[msg] = color
  end

  def error(msg)
    @error_msg = msg
  end

  def clear
    @error_msg = nil
    @echos = {}
  end
end

File: test/mocks/ui_mock.dart


import 'package:icli/utils/ui.dart';

class UIMock implements UI {

  var echos = [];
  String? errors;

  UIMock() {
    clear();
  }

  @override
  echo(String? text, Color color) {
    echos.add(text);
  }

  @override
  error(String msg) {
    errors = msg;
  }

  void clear() {
    echos = [];
    errors = null;
  }
}